pkg/mahttp/joinpath.go

```diff
@@ -0,0 +1,46 @@
+package mahttp
```

Comment:
How would we feel about putting this in `lib/` instead of it's own `mahttp` package?

Reply:
Whoops, you're right one doesn't exist yet. Would you mind creating one and moving this logic there? Over in https://github.com/storacha/piri/pull/169 I have done something similar - defined a `lib/` dir and started moving common logic over to it.